src/library/scala/collection/immutable/VectorMap.scala

```diff
@@ -106,7 +106,9 @@ final class VectorMap[K, +V] private[immutable] (
   // Only care about content, not ordering for equality
   override def equals(that: Any): Boolean =
     that match {
-      case vmap: VectorMap[_, _] => underlying == vmap.underlying
+      case vmap: VectorMap[_, _] =>
+        underlying.size == vmap.underlying.size &&
+          underlying.mapValues(_._2).toMap == vmap.underlying.mapValues(_._2).toMap
```

Comment:
Is there a way we can implement this without all these allocations? We can probably emplore a specialized version of the tricks used in Map#equals for something like
```scala
  override def equals(that: Any): Boolean =
    that match {
      case vmap: VectorMap[K, _] =>
        (this eq vmap) ||
          (this.size == vmap.size) && {
            try {
              var i = 0
              val _size = size
              while (i < _size) {
                val k = fields(i)
                if (!(vmap.getOrElse(k, VectorMap.Sentinel) == underlying(k)._2)) {
                  return false
                }
                i += 1
              }
              true
            } catch {
              case _: ClassCastException => false
            }
          }

      case _ => super.equals(that)
    }
```

Reply:
Yeah I know that there are faster methods, but it takes a while to compare them and benchmark them. I think that a vector iterator is probably going to be the fastest, but everything needs to be benched

Reply:
I don't think we should get it correct first and then get it fast.  This is too slow.  Let's get it correct and fast-ish first; leaving an implementation in that we are sure we won't want is a bad idea: it's just going to give us feedback that it's slow, and won't help us with correctness because it's going to be thrown away.

The way to test equality with _any_ map, not just `VectorMap`, is to make sure the sizes match, then iterate through all key-value pairs in `this` and make sure that the lookup in the other map is successful, and that the value returned is the same.  (Basically what Josh wrote except you should only pick out `Map[_, _]`.)

This, I think, is the baseline for correctness without anything too stupid.  (We should check to make sure that the super implementation doesn't already do this--I would have thought that deferring to the superclass implementation would have been adequate!)

Reply:
@Ichoran My earlier statement didn't mean to say that its not possible, just that I wanted to make sure that `equals` at least works correctly in the Scala 2.13.x branch.

I will set up a later PR that will do a faster equality, I am just a bit scrapped for time right now hence why I wanted to quickly fix it